bin/civihr-install

```diff
@@ -0,0 +1,823 @@
+        DB_USER="$1"
+        shift
+        ;;
+
+      --dbpass)
+        echo -n "Enter Password for Mysql:"
+        read -s DB_PASS 
+        echo 
+        DB_PASS="$1"
+        shift
+        ;;
+
+      --dbport)
+        DB_PORT="$1"
+        shift
+        ;;
+
+      --cmsdb)
+        [ "${1::1}" == "-" ] && echo "[[Missing value for option --cmsdb.]]" && civihr_app_usage && exit 1
+        CMSdbName="$1"
+        shift
+        ;;
+
+      --crmdb)
+        [ "${1::1}" == "-" ] && echo "[[Missing value for option --crmdb.]]" && civihr_app_usage && exit 1
+        CIVIdbName="$1"
+        shift
+        ;;
+
+      --url)
+        CMS_URL="$1"
+        shift
+        ;;
+
+      *)
+         if [ "${OPTION::1}" == "-" ]; then
+           echo "[[Unrecognized option: $OPTION]]"
+           civihr_app_usage
+           exit
+         fi
+        ;;
+    esac
+  done
+
+# Checking for all required options/arguments.
+[ -z "$DB_USER" ] || [ -z "$CMSdbName" ] || [ -z "$CIVIdbName" ] || [ -z "$CMS_URL" ] && echo "[[Please provide missing options/agruments.]]" && civihr_app_usage && exit 99
+
+# Assigning values.
+[ -z "$SITE_NAME" ]          && civihr_app_usage && exit 99
+[ -z "$WEB_ROOT" ]           && WEB_ROOT="$BLDDIR/$SITE_NAME"
+[ -z "$CMS_ROOT" ]           && CMS_ROOT="$WEB_ROOT"
+[ -z "$CIVI_SITE_KEY" ]      && CIVI_SITE_KEY=$(civihr_makepasswd 16)
+[ -z "$ADMIN_PASS" ]         && ADMIN_PASS=$(civihr_makepasswd 12)
+[ -z "$CMS_TITLE" ]          && CMS_TITLE="$SITE_NAME"
+[ -z "$SITE_TOKEN" ]         && SITE_TOKEN=$(civihr_makepasswd 16)
+
+}
+
+function absdirname() {
+  pushd $(dirname "$0") >> /dev/null
+    pwd
+  popd >> /dev/null
+}
+
+BINDIR=$(absdirname "$0")
+PRJDIR=$(pwd)
+TMPDIR="$PRJDIR/temp"
+BLDDIR="$PRJDIR"
+
+export PATH="$PATH:$BINDIR"
+civihr_parse "$@"
+
+# Checking for Drush dependencies if not present downloading into temp.
+command -v drush >/dev/null 2>&1 || ( echo "[[Downloading dependencies.........]]" && curl -Lo "$TMPDIR/drush" "https://github.com/drush-ops/drush/releases/download/8.1.8/drush.phar" && chmod +x "$TMPDIR/drush" && export PATH="$TMPDIR:$PATH" )
+
+###############################################################################
+## Downloads Drupal,CiviCRM and CiviHR from remote drush.make file.
+function civihr_setup_structure() {
+
+  [ -z "$CMS_VERSION" ] && CMS_VERSION=7.x
+  [ -z "$CIVI_VERSION" ] && CIVI_VERSION=4.4
+  [ -z "$HR_VERSION" ] && HR_VERSION=master
+
+  MAKEFILE="${TMPDIR}/${SITE_NAME}/${SITE_ID}.make"
+  civihr_makeparent "$MAKEFILE"
+  curl https://raw.githubusercontent.com/civicrm/civihr/PCHR-1666/bin/drush.make \
+    | sed "s;%%CACHE_DIR%%;${CACHE_DIR};" \
+    | sed "s;%%CIVI_VERSION%%;${CIVI_VERSION};" \
+    | sed "s;%%CMS_VERSION%%;${CMS_VERSION};" \
+    | sed "s;%%HR_VERSION%%;${HR_VERSION};" \
+    > "$MAKEFILE"
+
+  drush -y make --concurrency=5 --working-copy "$MAKEFILE" "$WEB_ROOT"
+}
+
+
+# Creates the default CiviHR users
+function create_default_users() {
```

Comment:
This function is civihr specific, shouldn't it be named `civihr_create_default_users`? (also applies to the other functions below which don't have the civihr prefix)